src/thread.rs

```diff
@@ -214,14 +221,10 @@ pub struct ScopedJoinHandle<'a, T: 'a> {
 /// crossbeam_utils::thread::scope(|scope| {
 ///     scope.defer(|| println!("Exiting scope"));
 ///     scope.spawn(|| println!("Running child thread in scope"))
-/// });
-/// // Prints messages in the reverse order written
```

Comment:
Is this not true anymore?

Reply:
It's true actually, but the methods spec doesn't guarantee it.  On the contrary to what is commented before, there's no order between joining spawned threads and running deferred functions.  I just removed the comment, effectively leave the order unspecified.

Definitely, we need to discuss what's the desired specification.  Asking the same question you raised before, what's the purpose of `defer()`?  What do we expect from it?  I think need to answer those questions before releasing a new version.  Of course we have to write tests (#12).

Reply:
This is the only use of `defer` I could find:
https://github.com/wayslog/statsd/blob/f27113df8667407b247d0ff6e63396a54c612e76/src/sender/mod.rs#L18

But looks like it's not necessary at all - calls to `.join()` can be invoked without `defer` as well.

I guess `defer` is actually useless and the same effect can be achieved using [`scopeguard::defer!`](https://docs.rs/scopeguard/0.3.3/scopeguard/macro.defer.html) anyway. So my suggestion would be to remove it.